docker-app/qfieldcloud/core/models.py

```diff
@@ -2374,16 +2374,18 @@ def for_user_and_project(self, user: User, project: Project) -> SecretQueryset:
             check_user_has_project_roles,
         )
 
-        if project.owner.is_organization:
-            # ensure the user has access to the organization
+        # ensure the user has access to the organization if private project
+        if project.owner.is_organization and not project.is_public:
             organization = Organization.objects.get(id=project.owner_id)
             check_user_has_organization_roles(
                 user, organization, OrganizationMember.ALL_ROLES
             )
         else:
             organization = None
-        # ensure the user has access to the project
-        check_user_has_project_roles(user, project, ProjectCollaborator.ALL_ROLES)
+
+        # ensure the user has access to the project if not public
+        if not project.is_public:
+            check_user_has_project_roles(user, project, ProjectCollaborator.ALL_ROLES)
```

Comment:
```suggestion
        check_user_has_project_roles(user, project, ProjectCollaborator.ALL_ROLES)
```

Isn't it sufficient without the explicit if statement? The user should have by default a "reader" role?

Reply:
Interesting, makes me consider changing the approach here.

Reason behind this if is this log line, spotted sometimes on public projects: `raise UserProjectRoleError(\nqfieldcloud.core.permissions_utils.UserProjectRoleError: User \"USER\" has not role \"['ADMIN', 'MANAGER', 'EDITOR', 'REPORTER', 'READER']\" on project PUBLIC_PROJECT."}`

Reply:
Addressed, thanks.